src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NFloat.cs

```diff
@@ -13,65 +17,691 @@
+        [NonVersionable]
+        public static NFloat operator -(NFloat left, NFloat right) => new NFloat(left._value - right._value);
+
+        /// <summary>Multiplies two values together to compute their product.</summary>
+        /// <param name="left">The value which <paramref name="right" /> multiplies.</param>
+        /// <param name="right">The value which multiplies <paramref name="left" />.</param>
+        /// <returns>The product of <paramref name="left" /> divided-by <paramref name="right" />.</returns>
+        [NonVersionable]
+        public static NFloat operator *(NFloat left, NFloat right) => new NFloat(left._value * right._value);
+
+        /// <summary>Divides two values together to compute their quotient.</summary>
+        /// <param name="left">The value which <paramref name="right" /> divides.</param>
+        /// <param name="right">The value which divides <paramref name="left" />.</param>
+        /// <returns>The quotient of <paramref name="left" /> divided-by <paramref name="right" />.</returns>
+        [NonVersionable]
+        public static NFloat operator /(NFloat left, NFloat right) => new NFloat(left._value / right._value);
+
+        /// <summary>Divides two values together to compute their remainder.</summary>
+        /// <param name="left">The value which <paramref name="right" /> divides.</param>
+        /// <param name="right">The value which divides <paramref name="left" />.</param>
+        /// <returns>The remainder of <paramref name="left" /> divided-by <paramref name="right" />.</returns>
+        [NonVersionable]
+        public static NFloat operator %(NFloat left, NFloat right) => new NFloat(left._value % right._value);
+
+        //
+        // Comparisons
+        //
+
+        /// <summary>Compares two values to determine equality.</summary>
+        /// <param name="left">The value to compare with <paramref name="right" />.</param>
+        /// <param name="right">The value to compare with <paramref name="left" />.</param>
+        /// <returns><c>true</c> if <paramref name="left" /> is equal to <paramref name="right" />; otherwise, <c>false</c>.</returns>
+        [NonVersionable]
+        public static bool operator ==(NFloat left, NFloat right) => left._value == right._value;
+
+        /// <summary>Compares two values to determine inequality.</summary>
+        /// <param name="left">The value to compare with <paramref name="right" />.</param>
+        /// <param name="right">The value to compare with <paramref name="left" />.</param>
+        /// <returns><c>true</c> if <paramref name="left" /> is not equal to <paramref name="right" />; otherwise, <c>false</c>.</returns>
+        [NonVersionable]
+        public static bool operator !=(NFloat left, NFloat right) => left._value != right._value;
+
+        /// <summary>Compares two values to determine which is less.</summary>
```

Comment:
Nit: This description seems a bit misleading, as it tells you whether left is less than right but not whether right is strictly less then left.

Consider adopting the existing doc terminology that float/double use, e.g. "Returns a value that indicates whether a specified Double value is less than or equal to another specified Double value."

Reply:
> Consider adopting the existing doc terminology

I always forget that we have APIs exposed here that are functionally unavailable and will never be called 👍 

I'll update this in a follow up PR as I need to also update `IComparisonOperators` and a few other places accordingly.